glogger/extra.py

```diff
@@ -0,0 +1,17 @@
+import logging
+from typing import Any, MutableMapping, Tuple
+
+
+class ExtraAdapter(logging.LoggerAdapter):
+    """
+    Logging adapter that adds an attribute named `_extra` containing a dict of all extra fields of a record.
+    This allows referring to the extras dict in a Formatter, when the extra field names differ among records.
+    """
+
+    def process(self, msg: Any, kwargs: MutableMapping[str, Any]) -> Tuple[Any, MutableMapping[str, Any]]:
```

Comment:
Would add the logic from maestro and from gProfiler that treats certain whitelist keywords differently.